src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIService.java

```diff
@@ -105,18 +141,36 @@ private void setupGitLabCIConfiguration(VcsRepositoryUrl repositoryURL, Programm
         }
 
         try {
-            // TODO: Update variables
-            projectApi.createVariable(repositoryPath, VARIABLE_DOCKER_IMAGE_NAME,
-                    programmingLanguageConfiguration.getImage(exercise.getProgrammingLanguage(), Optional.ofNullable(exercise.getProjectType())), Variable.Type.ENV_VAR, false,
-                    true);
-            projectApi.createVariable(repositoryPath, VARIABLE_BRANCH_NAME, exercise.getBranch(), Variable.Type.ENV_VAR, false, true);
-            projectApi.createVariable(repositoryPath, VARIABLE_ARTEMIS_SERVER_URL_NAME, artemisServerUrl.toExternalForm(), Variable.Type.ENV_VAR, false, true);
+            // TODO: Reduce the number of API calls
+
+            updateVariable(repositoryPath, VARIABLE_BUILD_DOCKER_IMAGE_NAME,
+                    programmingLanguageConfiguration.getImage(exercise.getProgrammingLanguage(), Optional.ofNullable(exercise.getProjectType())), true);
+            updateVariable(repositoryPath, VARIABLE_BUILD_LOGS_FILE_NAME, "build.log", true);
+            updateVariable(repositoryPath, VARIABLE_BUILD_PLAN_ID_NAME, buildPlanId, true);
+            updateVariable(repositoryPath, VARIABLE_CUSTOM_FEEDBACK_DIR_NAME, "TODO", false); // TODO
```

Comment:
Is this TODO-Item an oversight or will there be another followup?

Reply:
This is currently listed as future work, so probably there will be a followup.

Reply:
Can you remove or comment the `updateVariable()` call and add a more descriptive TODO comment?